nncf/quantization/algorithms/min_max/algorithm.py

```diff
@@ -335,42 +338,96 @@ def _get_quantization_target_points(self, model: TModel) -> OrderedDict[TargetPo
                 raise RuntimeError('Incorrect quantization point')
         self._quantization_target_points_to_qconfig = collections.OrderedDict(
             sorted(self._quantization_target_points_to_qconfig.items()))
         return self._quantization_target_points_to_qconfig
 
+    def _collect_unified_groups(self, quantizer_setup: SingleConfigQuantizerSetup) -> List[List[TargetPoint]]:
+        """
+        Collects the group of quantizers for unification.
+
+        :param quantizer_setup: SingleConfigQuantizerSetup instance
+        :return: List with the groups of the TargetPoints.
+        """
+        unified_scale_groups = []
+        for quantizer_ids in quantizer_setup.unified_scale_groups.values():
+            unified_scale_group = []
+            for quantizer_id in quantizer_ids:
+                quantization_point = quantizer_setup.quantization_points[quantizer_id]
+
+                # Only activation quantizers can be unified
+                if quantization_point.is_activation_quantization_point():
+                    activation_target_point = self._get_activation_quantization_target_point(quantization_point)
+                    unified_scale_group.append(activation_target_point)
+                else:
+                    raise RuntimeError('Incorrect quantization point')
+            unified_scale_groups.append(unified_scale_group)
+        return unified_scale_groups
+
+    def _get_graph_pattern(self, model: TModel) -> GraphPattern:
+        """
+        Returns full graph pattern for quantizer setup calculation.
+
+        :param model: Backend-specific model.
+        :return: GraphPattern instance.
+        """
+        backend = get_backend(model)
+        device = self._parameters.target_device
+        return PatternsManager.get_full_pattern_graph(backend, device)
+
     def _apply(self,
                model: TModel,
                statistic_points: Optional[StatisticPointsContainer] = None,
                dataset: Optional[Dataset] = None) -> TModel:
-        transformation_layout, transformation_commands = TransformationLayout(), []
+        transformation_layout = TransformationLayout()
         nncf_graph = NNCFGraphFactory.create(model) if self.nncf_graph is None else self.nncf_graph
         model_transformer = ModelTransformerFactory.create(model)
 
-        quantization_target_points = self._get_quantization_target_points(model)
+        pattern = self._get_graph_pattern(model)
+        quantizer_setup = self._get_quantizer_setup(nncf_graph, pattern)
+        quantization_target_points = self._get_quantization_target_points(quantizer_setup, nncf_graph)
+        unified_scale_groups = self._collect_unified_groups(quantizer_setup)
         weight_layer_names = set()
         def filter_func(point: StatisticPoint) -> bool:
             return MinMaxQuantization in point.algorithm_to_tensor_collectors and \
                    point.target_point == quantization_target_point
 
+        unified_ops_list = []
```

Comment:
Can we use a set here?

Reply:
You also check whether the `quantization_target_point` is in `unified_ops_list`

Reply:
I am not sure whether it is applicable here, but with a set, you should have search of elements for a constant time.